core/ts.core/src/ts/utils/ZipUtils.java

```diff
@@ -166,9 +168,13 @@ public static void extractTar(File file, File destination) throws IOException {
 
 					// Close the stream
 					out.close();
-					if (extractedFile.getParent().contains(BIN_FOLDER)) {
-						extractedFile.setExecutable(true);
-					}
+					// Preserve original modification date
+					extractedFile.setLastModified(entry.getTime());
+					long mode = entry.getMode(); 
+					if ((mode & 00100) > 0) { 
+						// Preserve execute permissions 
+						extractedFile.setExecutable(true, (mode & 00001) == 0); 
+					} 
 					break;
 				case TarEntry.LINK:
 					File linkFile = new File(destination, outFilename);
```

Comment:
Not sure.
For hard links It seems that is not necessary as It should keep original (linked) file permissions.
As for symbolic links, by default, It also seems that is not necessary (I checked with node-v6.9.4-linux-x64.tar.gz and npm symlink has all permissions -777-...).   Surely we could/sould do something like:

Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwxr-xr--");
FileAttribute<Set<PosixFilePermission>> attr = PosixFilePermissions.asFileAttribute(perms);
Files.createSymbolicLink(symLinkFile.toPath(), symTarget, attr);

but before that, we should check that current system is POSIX and "gess" permission pattern (in the example "rwxr-xr--") from entry attributes.  (I'm not sure but I think that if system is not POSIX and you try to pass those attrs an Exception is thrown).

Have you thought on using Apache Commons Compress to avoid this low level implementation?